File: 2460. Apply Operations to an Array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        for i in range(len(nums) - 1):
            if nums[i] == nums[i + 1]:
                nums[i + 1] = 0
                nums[i] *= 2

        ans = []
        count_zero = 0
        for num in nums:
            if num: ans.append(num)
            else: count_zero += 1
        
        while count_zero:
            ans.append(0)
            count_zero -= 1
        
        return ans

File: Array/2460. Apply Operations to an Array.cpp
class Solution {
    public:
        vector<int> applyOperations(vector<int>& nums) 
        {
            int n = nums.size();
            for(int i=0;i<n-1;i++)
            {
                if(nums[i]==nums[i+1])
                {
                    nums[i++] *= 2;
                    nums[i] = 0;
                }
            }
    
            vector<int>ans;
            for(int v:nums)
                if(v) ans.push_back(v);
            
            int zero = count(nums.begin(), nums.end(), 0);
            while(zero--) ans.push_back(0);
    
            return ans;
        }
    };